packages/advanced-logic/src/advanced-logic.ts

```diff
@@ -56,26 +56,30 @@ export default class AdvancedLogic implements AdvancedLogicTypes.IAdvancedLogic
     erc20TransferableReceivable: Erc20TransferableReceivablePaymentNetwork;
   };
 
+  private currencyManager: ICurrencyManager;
+
   constructor(currencyManager?: ICurrencyManager) {
     if (!currencyManager) {
       currencyManager = CurrencyManager.getDefault();
     }
+
+    this.currencyManager = currencyManager;
```

Comment:
I would go even further and force the injection at that level. This will enable removing the dependency from advanced-logic to the `currency` package (more in https://github.com/RequestNetwork/requestNetwork/pull/1238)

```suggestion
  constructor(private currencyManager: ICurrencyManager) {
```

Reply:
If you prefer, leave it like this and I'll update in my PR